File: app/src/main/java/knf/kuma/animeinfo/AnimeTagsAdapter.java
package knf.kuma.animeinfo;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import butterknife.BindView;
import butterknife.ButterKnife;
import knf.kuma.R;
import knf.kuma.search.GenreActivity;

public class AnimeTagsAdapter extends RecyclerView.Adapter<AnimeTagsAdapter.TagHolder>{

    private Context context;
    private List<String> list;

    public AnimeTagsAdapter(Context context, List<String> list) {
        this.context = context;
        this.list = list;
    }

    @NonNull
    @Override
    public TagHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new TagHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_chip,parent,false));
    }

    @Override
    public void onBindViewHolder(@NonNull final TagHolder holder, int position) {
        holder.chip.setText(list.get(position));
        holder.chip.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                GenreActivity.open(context, list.get(holder.getAdapterPosition()));
            }
        });

    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    class TagHolder extends RecyclerView.ViewHolder{
        @BindView(R.id.chip)
        TextView chip;

        TagHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this,itemView);
        }
    }
}

File: app/src/main/java/knf/kuma/animeinfo/AnimeTagsAdapter.kt
package knf.kuma.animeinfo

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import knf.kuma.R
import knf.kuma.search.GenreActivity
import org.jetbrains.anko.find

internal class AnimeTagsAdapter(private val context: Context, private val list: MutableList<String>) : RecyclerView.Adapter<AnimeTagsAdapter.TagHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TagHolder {
        return TagHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_chip, parent, false))
    }

    override fun onBindViewHolder(holder: TagHolder, position: Int) {
        holder.chip.text = list[position]
        holder.chip.setOnClickListener { GenreActivity.open(context, list[holder.adapterPosition]) }

    }

    override fun getItemCount(): Int {
        return list.size
    }

    internal inner class TagHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var chip: TextView = itemView.find(R.id.chip)
    }
}
